napari/_qt/widgets/qt_dims_slider.py

```diff
@@ -488,9 +492,11 @@ class QtPlayButton(QPushButton):
 
     play_requested = Signal(int)  # axis, fps
 
-    def __init__(self, dims, axis, reverse=False, fps=10, mode=LoopMode.LOOP):
+    def __init__(
+        self, qt_dims, axis, reverse=False, fps=10, mode=LoopMode.LOOP
+    ):
         super().__init__()
```

Comment:
I don't understand why this doesn't have `qt_dims` as a parent. And I wonder if this might be an alternative fix to holding a weak reference to `QtDims`.

If I understand the QObject lifetime model correctly, I think that if the child object is still alive, then its parent must also be (i.e. the parent effectively owns the child and is controls its lifetime/destruction). So it would not be possible for `_on_click` to be executed with a valid reference to `QtDims`.

More generally, I wonder how many of our object lifetime issues are caused by relying on Python references to Qt objects, rather than relying on the QObject hierarchy. Though that discussion is far beyond the scope of this PR :)

Reply:
Ah, I guess when we add the `QtPlayButton` to the layout, then reparenting will occur. So this doesn't seem like a viable alternative fix. Though I still can't explain why we need the weak reference.

Reply:
It's magic for me as well, but usually I found that weakreferences, even if not strictly necessary seem to help to have at least consistant crashes, or make it clearer when doing refactor who is the owner, and simplify reference graph when using objgraph. 

Something I haven't done there is also use wekref's callbacks to disconnect things but I don't think it's strictly necessary here.

I tend to [prefer to know reliably I'm lost](http://thecodelesscode.com/case/225).

Reply:
I didn't read your PR description properly, but after reading this message and thinking a bit more, the weak reference is justifiable (regardless of any bugs it does or doesn't cause) because it breaks a cyclic reference between Python objects (specifically, `QtDims` and `QtPlayButton` with `QtDimSliderWidget` in the middle of the cycle). Might be good to add that as an explanatory implementation comment to the code to help the next lost soul.

In general, my guess is there's only so much pyqt/pyside can do when it comes to keeping Python and QObject lifetimes consistent/safe. Would be great if anyone knows of a resource that explains how to use those more safely (other than reading through the code itself).

Anyway, I'm pretty close to approving this. I just need to think through the dropped reference to `QtDimSliderWidget.thread`.